pallets/ajuna-awesome-avatars/src/tests.rs

```diff
@@ -3987,3 +3987,227 @@ mod ipfs {
+		ExtBuilder::default()
+			.balances(&[(ALICE, initial_balance)])
+			.affiliators(&[ALICE])
+			.build()
+			.execute_with(|| {
+				assert_eq!(
+					pallet_ajuna_affiliates::Affiliators::<Test, AffiliatesInstance1>::get(ALICE),
+					AffiliatorState { status: AffiliatableStatus::Affiliatable, affiliates: 0 }
+				);
+
+				assert_ok!(AAvatars::add_affiliation(RuntimeOrigin::signed(BOB), 0));
+
+				System::assert_last_event(mock::RuntimeEvent::Affiliates(
+					pallet_ajuna_affiliates::Event::AccountAffiliated { account: BOB, to: ALICE },
+				));
+
+				assert_eq!(
+					pallet_ajuna_affiliates::Affiliatees::<Test, AffiliatesInstance1>::get(BOB),
+					Some(bounded_vec![ALICE])
+				);
+
+				assert_eq!(
+					pallet_ajuna_affiliates::Affiliators::<Test, AffiliatesInstance1>::get(ALICE),
+					AffiliatorState { status: AffiliatableStatus::Affiliatable, affiliates: 1 }
+				);
+			});
+	}
+
+	#[test]
+	fn cannot_affiliate_to_non_enabled_account() {
+		let initial_balance = 1_000_000;
+		ExtBuilder::default()
+			.balances(&[(ALICE, initial_balance)])
+			.build()
+			.execute_with(|| {
+				assert_eq!(
+					pallet_ajuna_affiliates::Affiliators::<Test, AffiliatesInstance1>::get(ALICE),
+					AffiliatorState { status: AffiliatableStatus::NonAffiliatable, affiliates: 0 }
+				);
+
+				assert_noop!(
+					AAvatars::add_affiliation(RuntimeOrigin::signed(BOB), 0),
+					Error::<Test>::AffiliatorNotFound
+				);
+			});
+	}
+
+	#[ignore]
+	#[test]
+	fn enable_account_for_affiliation_free() {
+		let initial_balance = 1_000_000;
+		ExtBuilder::default()
+			.balances(&[(ALICE, initial_balance)])
+			.build()
+			.execute_with(|| {
+				assert_ok!(AAvatars::enable_affiliator(
+					RuntimeOrigin::signed(ALICE),
+					AffiliatorTarget::OneselfFree
+				));
+
+				assert_eq!(
+					pallet_ajuna_affiliates::Affiliators::<Test, AffiliatesInstance1>::get(ALICE),
+					AffiliatorState { status: AffiliatableStatus::Affiliatable, affiliates: 0 }
+				);
+				assert_eq!(
+					pallet_ajuna_affiliates::NextAffiliateId::<Test, AffiliatesInstance1>::get(),
+					1
+				);
+
+				assert_eq!(<Test as Config>::Currency::free_balance(ALICE), initial_balance);
+				assert_eq!(
+					<Test as Config>::Currency::free_balance(AAvatars::treasury_account_id()),
+					0
+				);
+
+				System::assert_last_event(mock::RuntimeEvent::Affiliates(
+					pallet_ajuna_affiliates::Event::AccountMarkedAsAffiliatable {
+						account: ALICE,
+						affiliate_id: 0,
+					},
+				));
+			});
+	}
+
+	#[test]
+	fn enable_account_for_affiliation_paying() {
+		let initial_balance = 1_000_000;
+		let affiliator_enable_fee = 1_000;
+		ExtBuilder::default()
+			.balances(&[(ALICE, initial_balance)])
+			.build()
+			.execute_with(|| {
+				GlobalConfigs::<Test>::mutate(|config| {
+					config.affiliate_config.affiliator_enable_fee = affiliator_enable_fee;
+				});
+
+				assert_ok!(AAvatars::enable_affiliator(
+					RuntimeOrigin::signed(ALICE),
+					AffiliatorTarget::OneselfPaying
```

Comment:
Yeah, I put Oneself because Self is a reserved keyword but I can use the `r#Self` syntax for it if its better.

It may not be in the docs but it can be found in this ticket here -> https://ajunanetwork.atlassian.net/browse/AJUN-187